cpp/unreal_plugins/SpEngine/Source/SpEngine/GameWorldService.h

```diff
@@ -41,6 +64,120 @@ class GameWorldService {
             SP_ASSERT(world_);
```

Comment:
We are currently using this function in while loops to make sure that the desired level has been loaded. But loading a level is a predictable operation, so we shouldn't need a while loop. And this will get stuck in an infinite loop in cases where the world can't be loaded, which is not as developer-friendly as an `assert`. We should replace these while loops with something more predictable, e.g.,

```
engine_service.begin_tick()
game_world_service.open_level(...)
engine_service.tick()
engine_service.end_tick()

engine_service.begin_tick()
engine_service.tick()
engine_service.end_tick()

assert game_world_service.get_world_name() == desired_level_name
```